subsys/usb/usb_device.c

```diff
@@ -612,14 +657,35 @@ static bool usb_set_interface(u8_t iface, u8_t alt_setting)
 				if_desc = (void *)p;
 			}
 
-			LOG_DBG("iface_num %u alt_set %u", iface, alt_setting);
+			LOG_DBG("Current iface %u alt setting %u",
+				cur_iface, cur_alt_setting);
 			break;
 		case USB_ENDPOINT_DESC:
-			if ((cur_iface != iface) ||
+			if ((cur_iface == iface) &&
```

Comment:
Maybe it would be more beautiful to move the entire block in a function `static bool usb_reset_iface_endpoints()`

```c
if ((cur_iface == iface) {
    usb_reset_iface_endpoints(....);
}
```

and set/reset endpoints there?